bundles/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/internal/handler/KeContactHandler.java

```diff
@@ -94,30 +97,49 @@ public KeContactHandler(Thing thing, KeContactTransceiver transceiver) {
-                    Math.max((((BigDecimal) getConfig().get(POLLING_REFRESH_INTERVAL)).intValue()) - 5, 0) * 1000);
-
-            cache.put(CACHE_REPORT_1, () -> transceiver.send("report 1", getHandler()));
-            cache.put(CACHE_REPORT_2, () -> transceiver.send("report 2", getHandler()));
-            cache.put(CACHE_REPORT_3, () -> transceiver.send("report 3", getHandler()));
-            cache.put(CACHE_REPORT_100, () -> transceiver.send("report 100", getHandler()));
-
-            if (pollingJob == null || pollingJob.isCancelled()) {
-                try {
-                    pollingJob = scheduler.scheduleWithFixedDelay(this::pollingRunnable, 0,
-                            ((BigDecimal) getConfig().get(POLLING_REFRESH_INTERVAL)).intValue(), TimeUnit.SECONDS);
-                } catch (Exception e) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
-                            "An exception occurred while scheduling the polling job");
+        logger.debug("Initializing Keba binding");
+        try {
+            if (isKebaReachable()) {
+                transceiver.registerHandler(this);
+
+                cache = new ExpiringCacheMap<>(
+                        Math.max((((BigDecimal) getConfig().get(POLLING_REFRESH_INTERVAL)).intValue()) - 5, 0) * 1000);
+
+                cache.put(CACHE_REPORT_1, () -> transceiver.send("report 1", getHandler()));
+                cache.put(CACHE_REPORT_2, () -> transceiver.send("report 2", getHandler()));
+                cache.put(CACHE_REPORT_3, () -> transceiver.send("report 3", getHandler()));
+                cache.put(CACHE_REPORT_100, () -> transceiver.send("report 100", getHandler()));
+
+                if (pollingJob == null || pollingJob.isCancelled()) {
+                    try {
+                        pollingJob = scheduler.scheduleWithFixedDelay(this::pollingRunnable, 0,
+                                ((BigDecimal) getConfig().get(POLLING_REFRESH_INTERVAL)).intValue(), TimeUnit.SECONDS);
+                    } catch (Exception e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                                "An exception occurred while scheduling the polling job");
+                    }
                 }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "IP address or port number not set");
             }
-        } else {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "IP address or port number not set");
+        } catch (Exception e) {
+            logger.error("Exception during Ping check and Http Backup check", e);
+        }
+    }
+
+    private boolean isKebaReachable() throws IOException {
+        boolean isReachable = false;
+        SocketAddress sockAddr = new InetSocketAddress((String) getConfig().get(IP_ADDRESS), SOCKET_CHECK_PORT_NUMBER);
```

Comment:
InetSocketAddress will return an `IllegalArgumentException` if the ip address is not set. Better check it beforehand whether it is null.

Reply:
Using getIPAddress() now. This function returns "" instead of null, if not configured (which is not possible, as the parameter is mandatory).
But even the null check is not sufficient. Even when != null it must not be a vaild IP address. The easiest way to figure that out is to try the init and catch the exception, isn't it?

Reply:
If it is not a valid IP address, the string will be interpreted as a host name. If the host name can't be resovled, `Socket.connect()` will throw an `IOException`.

The IP address can't be `null` if the user uses the UI to configure the binding. But there will be no sanity checks if the user uses textual configuration. If the user doesn't specify the IP address, it will be null.

Reply:
Thanks for the hint with the textual config

Reply:
added a get'er `getRefreshInterval()` that handles `null.`
`getRefreshInterval()` and `getIPAddress()` are now always used when accessing the config